webots_ros2_epuck/webots_ros2_epuck/epuck_node.py

```diff
@@ -46,47 +42,33 @@
 ]
 
 
-DEVICE_CONFIG = {
-    'camera': {'topic_name': ''},
-    'robot': {'publish_base_footprint': True},
-    'ps0': {'always_publish': True},
-    'ps1': {'always_publish': True},
-    'ps2': {'always_publish': True},
-    'ps3': {'always_publish': True},
-    'ps4': {'always_publish': True},
-    'ps5': {'always_publish': True},
-    'ps6': {'always_publish': True},
-    'ps7': {'always_publish': True},
-    'tof': {'always_publish': True}
-}
-
-
-class EPuckDriver(WebotsDifferentialDriveNode):
-    def __init__(self, args):
-        super().__init__(
-            'epuck_driver',
-            args,
-            wheel_distance=DEFAULT_WHEEL_DISTANCE,
-            wheel_radius=DEFAULT_WHEEL_RADIUS
-        )
-        self.start_device_manager(DEVICE_CONFIG)
+class EPuckNode(Node):
+    def __init__(self):
+        super().__init__('epuck_node')
+        self.get_logger().info("Epuck node has been started.")
 
         # Intialize distance sensors for LaserScan topic
-        self.distance_sensors = {}
+        self.__subscriber_dist_sensors = {}
+        self.__dists = {}
         for i in range(NB_INFRARED_SENSORS):
-            sensor = self.robot.getDistanceSensor('ps{}'.format(i))
-            sensor.enable(self.timestep)
-            self.distance_sensors['ps{}'.format(i)] = sensor
+            self.__dists['ps{}'.format(i)] = OUT_OF_RANGE
+
+        self.__subscriber_dist_sensors['ps0'] = self.create_subscription(Range, '/e_puck/ps0', self.__process_dist_sens_0, 1)
+        self.__subscriber_dist_sensors['ps1'] = self.create_subscription(Range, '/e_puck/ps1', self.__process_dist_sens_1, 1)
+        self.__subscriber_dist_sensors['ps2'] = self.create_subscription(Range, '/e_puck/ps2', self.__process_dist_sens_2, 1)
+        self.__subscriber_dist_sensors['ps3'] = self.create_subscription(Range, '/e_puck/ps3', self.__process_dist_sens_3, 1)
+        self.__subscriber_dist_sensors['ps4'] = self.create_subscription(Range, '/e_puck/ps4', self.__process_dist_sens_4, 1)
+        self.__subscriber_dist_sensors['ps5'] = self.create_subscription(Range, '/e_puck/ps5', self.__process_dist_sens_5, 1)
+        self.__subscriber_dist_sensors['ps6'] = self.create_subscription(Range, '/e_puck/ps6', self.__process_dist_sens_6, 1)
+        self.__subscriber_dist_sensors['ps7'] = self.create_subscription(Range, '/e_puck/ps7', self.__process_dist_sens_7, 1)
+
+        self.__subscriber_tof = self.create_subscription(Range, '/e_puck/tof', self.__process_tof, 1)
+        self.__new_tof_data = False
```

Comment:
We can treat it as the other distances sensors (we keep the value in `self.__distances`). If `self.__distances['tof'] == OUT_OF_RANGE` than the value will be automatically ignored, no special "treatment" is needed.

Reply:
Hum, so you would to put tof in `self.__distances['tof']`, but then it has to not go through the check `laser_dists = [OUT_OF_RANGE if dist > INFRARED_MAX_RANGE else dist for dist in dists]` as ToF has another `INFRARED_MAX_RANGE`, right ?